drivers/hmis/app/graphql/mutations/ce/mark_units_available.rb

```diff
@@ -24,15 +24,21 @@ def resolve(unit_ids:)
       project = Hmis::Hud::Project.find_by(id: project_ids.first)
       access_denied! unless current_user.permissions_for?(project, :can_manage_units)
 
-      # TODO(#7529) - template should be determined by context (project, unit type, ...)
-      # For now, if you are using the "starter pack," this picks the template that creates an enrollment
-      template = Hmis::WorkflowDefinition::Template.viewable_by(current_user).last
-      raise unless template.present?
+      # Determine which template to use for each unit based on Unit Group configuration.
+      # TODO(#7753) Once we finalize the frontend and adjust this mutation, we may want to:
+      # - Raise if called for units belonging to several unit groups (maybe allowed, tbd?)
+      unit_to_template = units.preload(:unit_group).map do |unit|
+        workflow_template = unit.unit_group&.workflow_template
+        raise 'Unable to mark unit available because there is no associated workflow template' unless workflow_template.present?
+
+        [unit.id, workflow_template]
+      end.to_h
```

Comment:
small adjustments:
* might as well preload workflow template
* present? shouldn't be needed
* can use to_h with a block rather than map
```suggestion
      unit_to_template = units.preload(unit_group: :workflow_template).to_h do |unit|
        workflow_template = unit.unit_group&.workflow_template
        raise 'Unable to mark unit available because there is no associated workflow template' unless workflow_template

        [unit.id, workflow_template]
      end
```